paint/PAINT_ColourPicker.cpp

```diff
@@ -18,32 +18,64 @@ ColourPicker::ColourPicker(gfx::Rectangle rect, SDL_Renderer* renderer, std::sha
 	, primaryColourDisplay_(std::make_shared<win::ColourDisplay>(gfx::Rectangle(), "primaryColourDisplay", drawWindow->getPrimaryRGBA(), renderer, true))
 	, secondaryColourDisplay_(std::make_shared<win::ColourDisplay>(gfx::Rectangle(), "secondaryColourDisplay",drawWindow-> getSecondaryRGBA(), renderer, false))
 	, swapButton_(std::make_shared<win::Button>(renderer, gfx::Rectangle(rect.x + 75, rect.y + 20, 30, 30), "swapColourButton", "button_swap_colours.png", paint::swapColours))
-	, colourValuesBox_(std::make_shared<Container>(std::make_shared<win::TableLayout>(0, 0, 0, 20, 4, 1), gfx::Rectangle(rect.x + rect.width - 50, rect.y + 65, 40, 140), "colourValuesBox"))
 	, swappedDisplays_(false)
+	, colourValuesBox_(std::make_shared<Container>(std::make_shared<win::TableLayout>(0, 0, 0, 20, 4, 1), gfx::Rectangle(rect.x + rect.width - 50, rect.y + 65, 40, 140), "colourValuesBox"))
+	, colourSliders_(std::make_shared<Container>(std::make_shared<win::TableLayout>(0, 0, 0, 20, 4, 1), gfx::Rectangle(rect.x+5, rect.y + 65, 120, 140), "colourSlidersBox"))
 {
 	addChild(swapButton_);
 	
+	initialiseColourDisplays();
+	initialiseColourValueBoxes();
+	initialiseColourSliders();
+	
+	setDrawColourFromActive();
+	updateColourSliders();
+}
+
+void ColourPicker::initialiseColourDisplays()
```

Comment:
Is this function called from more than one place? If not, best remove it as it needs to be in the header and that exposes client code to implementation detail. Here you should just inline the implementation, otherwise in some situations you can create a local (static) function to do the heavy lifting and pass the required data into it.
**SHOULD**

Reply:
moved these 'initialise' functions to private and made inline functions.